remappers/kubernetesmetrics/k8smetrics_test.go

```diff
@@ -0,0 +1,220 @@
+			{Type: Sum, Name: "k8s.pod.network.io", DP: testDP{Ts: now, Int: ptr(int64(1024)), Attrs: map[string]any{"device": DEVICE, "direction": "receive"}}},
+			{Type: Sum, Name: "k8s.pod.network.io", DP: testDP{Ts: now, Int: ptr(int64(2048)), Attrs: map[string]any{"device": DEVICE, "direction": "transmit"}}},
+		},
+	}
+
+	scopeMetrics := make([]pmetric.ScopeMetrics, 0, len(in))
+	for scraper, m := range in {
+		sm := pmetric.NewScopeMetrics()
+		sm.Scope().SetName(fmt.Sprintf("%s/%s", scopePrefix, scraper))
+		testMetricToMetricSlice(b, m, sm.Metrics())
+		scopeMetrics = append(scopeMetrics, sm)
+	}
+
+	r := NewRemapper(zaptest.NewLogger(b))
+	resource := pcommon.NewResource()
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		for _, sm := range scopeMetrics {
+			r.Remap(sm, pmetric.NewMetricSlice(), resource)
+		}
+	}
+}
+
+type testMetric struct {
+	Name string
+	Type pmetric.MetricType
+	DP   testDP
+}
+
+type testDP struct {
+	Ts    pcommon.Timestamp
+	Dbl   *float64
+	Int   *int64
+	Attrs map[string]any
+}
+
+func metricSliceToTestMetric(t *testing.T, ms pmetric.MetricSlice) []testMetric {
+	testMetrics := make([]testMetric, ms.Len())
+	for i := 0; i < ms.Len(); i++ {
+		m := ms.At(i)
+		testMetrics[i].Name = m.Name()
+		testMetrics[i].Type = m.Type()
+
+		var dps pmetric.NumberDataPointSlice
+		switch m.Type() {
+		case pmetric.MetricTypeGauge:
+			dps = m.Gauge().DataPoints()
+		case pmetric.MetricTypeSum:
+			dps = m.Sum().DataPoints()
+		}
+
+		if dps.Len() != 1 {
+			t.Fatalf("unexpected metric, test is written assuming each metric with a single datapoint")
+		}
+
+		dp := dps.At(0)
+		testMetrics[i].DP = testDP{Ts: dp.Timestamp(), Attrs: dp.Attributes().AsRaw()}
+		switch dp.ValueType() {
+		case pmetric.NumberDataPointValueTypeInt:
+			testMetrics[i].DP.Int = ptr(dp.IntValue())
+		case pmetric.NumberDataPointValueTypeDouble:
+			testMetrics[i].DP.Dbl = ptr(dp.DoubleValue())
+		}
+	}
+
+	return testMetrics
+}
+
+func testMetricToMetricSlice(t testing.TB, testMetrics []testMetric, out pmetric.MetricSlice) {
+	out.EnsureCapacity(len(testMetrics))
+
+	for _, testm := range testMetrics {
+		m := out.AppendEmpty()
+		m.SetName(testm.Name)
+
+		var dps pmetric.NumberDataPointSlice
+		switch typ := testm.Type; typ {
+		case pmetric.MetricTypeGauge:
+			dps = m.SetEmptyGauge().DataPoints()
+		case pmetric.MetricTypeSum:
+			dps = m.SetEmptySum().DataPoints()
+		default:
+			t.Fatalf("unhandled metric type %s", typ)
+		}
+
+		dp := dps.AppendEmpty()
+		dp.SetTimestamp(testm.DP.Ts)
+		if testm.DP.Int != nil {
+			dp.SetIntValue(*testm.DP.Int)
+		} else if testm.DP.Dbl != nil {
+			dp.SetDoubleValue(*testm.DP.Dbl)
+		}
+		if err := dp.Attributes().FromRaw(testm.DP.Attrs); err != nil {
+			t.Fatalf("failed to copy attributes from test data: %v", err)
+		}
+	}
+}
+
+func ptr[T any](v T) *T {
+	return &v
+}
```

Comment:
This is common to  testing all remappsers (since the metric package is common). I would suggest moving this bit into the `internal/metricstest` package or something similar and using it in both k8s and hostmetrics testing.

Reply:
Thanks @lahsivjar moved this common part to internal folder.
Hostmetrics tests updated too

Reply:
Sorry for so much back and forth. We should put the testing in a separate package for better isolation since, _I think_, the binary would have some definitions of the structs and other things if the package is referenced in a code. Better to just create a test utility package. Feel free to merge this PR and do it as a followup.